internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala

```diff
@@ -1086,20 +1090,29 @@ private final class AnalysisCallback(
 
   private def knownProducts(merged: Analysis) = {
     // List classes defined in the files that were compiled in this run.
-    val ps = java.util.concurrent.ConcurrentHashMap.newKeySet[String]
-    val knownProducts: ParVector[VirtualFileRef] =
-      new ParVector(merged.relations.allSources.toVector)
-        .flatMap(merged.relations.products)
-    // extract product paths in parallel
+    val ps: java.util.Set[String] =
+      if (AnalysisCallback.parallelKnownProducts)
+        java.util.concurrent.ConcurrentHashMap.newKeySet[String]
+      else
+        new java.util.HashSet[String]()
+    val knownProducts: Vector[VirtualFileRef] =
+      merged.relations.allSources.toVector.flatMap(merged.relations.products)
+
+    def knownProductsPar =
+      if (AnalysisCallback.parallelKnownProducts) new ParVector(knownProducts) else knownProducts
```

Comment:
Does this use the default fork/join scheduler? If so, could it end up competing against other tasks or potentially other Zinc compilation that are also trying to do the same thing? Or it finishes quickly enough that it won't surface? (But takes long enough time if it's not parallelized?)

Reply:
Looks like I added the parallel thing in  https://github.com/sbt/zinc/pull/849/commits/94ba81e908d3ec4838705125fbaee9156f2be496, but I'd be happy to see it gone if it's not working.

Reply:
In this case I think the best idea is to keep it sequential. If this is a bottle neck, we can try to first reduce the overhead in sequential code (like I've done here, but I'm sure there are more opportunities.)  Sequential code is a bit easier to profile and tune.

I've added a commit to remove the `ParVector` altogether.